internal/entities/properties/properties.go

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package properties provides a simple way to access properties of an entity
+package properties
+
+import (
+	"fmt"
+)
+
+// Property is a struct that holds a value. It's just a wrapper around an interface{}
+// with typed getters and handling of an empty receiver
+type Property struct {
+	value any
+}
+
+// NewProperty creates a new Property with a given value
+func NewProperty(value any) *Property {
+	return &Property{value: value}
+}
+
+func propertyValueAs[T any](value any) (T, error) {
+	var zero T
+	val, ok := value.(T)
+	if !ok {
+		return zero, fmt.Errorf("value is not of type %T", zero)
+	}
+	return val, nil
+}
+
+// AsBool returns the boolean value, or an error if the value is not a boolean
+func (p *Property) AsBool() (bool, error) {
+	if p == nil {
+		return false, fmt.Errorf("property is nil")
+	}
+	return propertyValueAs[bool](p.value)
+}
+
+// GetBool returns the boolean value, or false if the value is not a boolean
+func (p *Property) GetBool() bool {
```

Comment:
The idea behind having separate Get and As is that Get is a convenience function for when you're OK not having the data but just the default value for that type and As does return an error that you need to handle for critical values (repo name, repo id, ...)